lib/src/view/game/game_player.dart

```diff
@@ -140,16 +143,25 @@ class GamePlayer extends StatelessWidget {
           ),
         if (timeToMove != null)
           MoveExpiration(timeToMove: timeToMove!, mePlaying: mePlaying)
-        else if (materialDiff != null)
+        else if (materialDiff != null && materialDifference?.visible == true)
           Row(
             children: [
-              for (final role in Role.values)
-                for (int i = 0; i < materialDiff!.pieces[role]!; i++)
-                  Icon(
-                    _iconByRole[role],
-                    size: 13,
-                    color: Colors.grey,
-                  ),
+              if (materialDifference == MaterialDifference.capturedPieces)
```

Comment:
Maybe it would be more readable to factor out this whole `Row` into a `class MaterialDifferenceDisplay extends StatelessWidget`. (Then you could also precompute the pieces to loop over in a helper variable and use that in the loop, instead of needing to write the loop twice)